core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -178,10 +192,22 @@ private[chisel3] trait HasId extends InstanceId {
   }
 
   /** This resolves the precedence of [[autoSeed]] and [[suggestName]]
+    *
+    * @note It will become an error in the future to suggestName the same thing that autoSeed would have assigned
     *
     * @return the current calculation of a name, if it exists
     */
-  private[chisel3] def seedOpt: Option[String] = suggested_seed.orElse(auto_seed)
+  private[chisel3] def seedOpt: Option[String] = {
+    suggested_seed.zip(auto_seed).foreach {
+      case (suggested, auto) =>
+        if (suggested == auto) {
```

Comment:
```suggestion
    if (suggested_seed.isDefined && auto_seed.isDefined) {
      if (suggested_seed.get == auto_seed.get) {
```
I suggested this in a comment that got lost but I think this is worth micro-optimizing to avoid any allocations as this is called on every `Data`.